### PR TITLE
Add API integration to Server Submit

### PR DESCRIPTION
**Pull request by Oksamies**
Server listings can now be added

### CODE REVIEW

**Review by MythicManiac — APPROVED** [review]

File: apps/sertra/pages/create.tsx [hunk 1]
```diff
@@ -17,13 +17,45 @@ export const getStaticProps: GetStaticProps = async () => {
...
+    formState: { errors },
+  } = useForm<Inputs>();
+  const onSubmit: SubmitHandler<Inputs> = (data) => {
+    const prepped_data = {
+      name: data.serverName,
+      description: data.description,
+      community: "v-rising",
+      connection_data: data.connectionInfo,
+      // TODO: Populate this with mod ids etc
+      mods: ["1", "2", "3"],
```
Comment (VilppeRiskidev) [inline apps/sertra/pages/create.tsx:44]: When is tihs TODO going to be done and/or is it blocked by something?

File: apps/sertra/pages/create.tsx [hunk 2]
```diff
@@ -39,37 +71,43 @@ const SubmitServer: NextPage<{ swrFallback: { [key: string]: unknown } }> = ({
...
+            <select className={styles.input} id={"mode"} {...register("mode")}>
               <option value={"pvp"}>PvP</option>
               <option value={"pve"}>PvE</option>
             </select>
           </div>
           <div className={styles.row}>
             <label htmlFor={"mods"}>Mods</label>
-            <select className={styles.input} id={"mods"}>
+            <select className={styles.input} id={"mods"} {...register("mods")}>
               {/*TODO: Populate*/}
```
Comment (VilppeRiskidev) [inline apps/sertra/pages/create.tsx:111]: When is this TODO going to be done and/or is it blocked by something?

**Comment by VilppeRiskidev** [comment]
added a couple of questions, but if those have reasonable answers, it is good to go 👍

File: apps/sertra/package.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
     "next": "12.1.6",
     "react": "18.1.0",
     "react-dom": "18.1.0",
-    "react-hook-form": "^7.32.2",
+    "react-hook-form": "^7.34.2",
```
Comment (anttimaki) [inline apps/sertra/package.json:20]: The commit doesn't include a updated `yarn.lock` file. You should run `yarn install`.

File: apps/sertra/pages/create.tsx [hunk 4]
```diff
@@ -1,10 +1,10 @@
 import { GetStaticProps, NextPage } from "next";
-import { SWRConfig } from "swr";
-
+import { ApiURLs, TsApiURLs } from "../api/urls";
 import { getPackageList } from "../api/hooks";
-import { TsApiURLs } from "../api/urls";
+import { SWRConfig } from "swr";
```
Comment (anttimaki) [inline apps/sertra/pages/create.tsx:4]: Please don't mess up [the import order](https://github.com/thunderstore-io/thunderstore-ui/pull/510/commits/618a1b07c383e9645e0652d83c0bbd28c89b6725).

File: apps/sertra/pages/create.tsx [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
 import { ModSelectorModal } from "../components/mod-selector/modal";
 import styles from "../styles/SubmitServer.module.css";
+import { useForm, SubmitHandler } from "react-hook-form";
```
Comment (anttimaki) [inline apps/sertra/pages/create.tsx:7]: Import order.

File: apps/sertra/pages/create.tsx [hunk 6]
```diff
@@ -17,13 +17,45 @@ export const getStaticProps: GetStaticProps = async () => {
   };
 };
 
+type Inputs = {
+  serverName: string;
+  connectionInfo: string;
+  description: string;
+  mode: string;
+  mods: [string];
+  isPasswordProtected: string;
```
Comment (anttimaki) [inline apps/sertra/pages/create.tsx:26]: Should this be boolean? Or did RHF handle everything as stings?

File: apps/sertra/pages/create.tsx [hunk 7]
```diff
@@ -17,13 +17,45 @@ export const getStaticProps: GetStaticProps = async () => {
...
+
 const SubmitServer: NextPage<{ swrFallback: { [key: string]: unknown } }> = ({
   swrFallback,
 }) => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<Inputs>();
+  const onSubmit: SubmitHandler<Inputs> = (data) => {
```
Comment (anttimaki) [inline apps/sertra/pages/create.tsx:37]: Code style nitpick, empty line here would be nice.

File: apps/sertra/pages/create.tsx [hunk 8]
```diff
@@ -39,37 +71,43 @@ const SubmitServer: NextPage<{ swrFallback: { [key: string]: unknown } }> = ({
             <input
               className={styles.input}
               id={"serverName"}
-              name={"serverName"}
               type={"text"}
+              {...register("serverName", { required: true })}
             />
+            {/* Add styling here */}
```
Comment (anttimaki) [inline apps/sertra/pages/create.tsx:77]: Should this be handled now? Or at least add TODO to make these easier to find in the future.

File: apps/sertra/pages/create.tsx [hunk 9]
```diff
@@ -39,37 +71,43 @@ const SubmitServer: NextPage<{ swrFallback: { [key: string]: unknown } }> = ({
...
             <label htmlFor={"connectionInfo"}>Connection Info</label>
             <input
               className={styles.input}
-              name={"connectionInfo"}
               id={"connectionInfo"}
               type={"text"}
+              {...register("connectionInfo", { required: true })}
             />
+            {/* Add styling here */}
+            {errors.serverName?.type === "required" &&
```
Comment (anttimaki) [inline apps/sertra/pages/create.tsx:90]: Should this be `errors.connectionInfo`?

**Review by anttimaki — CHANGES_REQUESTED** [review]

File: apps/sertra/pages/create.tsx (hunk 1 above)
Comment (Oksamies) [reply apps/sertra/pages/create.tsx:44]: The populating is blocked by TS-712, by not having the actual mod data available. This is going to need an separate task, so I'm adding it to the inbox.

Format of the mods is currently:
`"mods":["ebkr-r2modman-3.1.31","Mythic-ServerLaunchFix-0.1.0","molenzwiebel-Wetstone-1.1.0","BepInEx-BepInExPack_V_Rising-1.0.0","ModLoader64-HuntControl-1.1.1","iZastic-AutoReady-1.0.1","iZastic-RemoveVignette-1.0.0","RPGMods-RPGMods-0.0.3"]`
I was just lazy and put 1 2 3 there 👍

File: apps/sertra/pages/create.tsx (hunk 4 above)
Comment (Oksamies) [reply apps/sertra/pages/create.tsx:4]: Oof looks like the merge switched it around, I'll fix.

File: apps/sertra/pages/create.tsx (hunk 8 above)
Comment (Oksamies) [reply apps/sertra/pages/create.tsx:77]: Yeah, I'm not sure what is the correct styling for this. Just a red border warning or an actual message 🤷 I'll add a ticket about this.

File: apps/sertra/pages/create.tsx (hunk 9 above)
Comment (Oksamies) [reply apps/sertra/pages/create.tsx:90]: Yes 👍

File: apps/sertra/pages/create.tsx (hunk 8 above)
Comment (MythicManiac) [reply apps/sertra/pages/create.tsx]: > I'm not sure what is the correct styling for this.
Just have the comment start with `TODO:`, doesn't matter what the wrapping syntax is

File: apps/sertra/pages/create.tsx (hunk 5 above)
Comment (anttimaki) [reply apps/sertra/pages/create.tsx:7]: RHF should be up there with the rest of 3rd party libraries.

File: apps/sertra/pages/create.tsx (hunk 1 above)
Comment (anttimaki) [reply apps/sertra/pages/create.tsx:46]: Not sure if the reply is for me or Vilppe or both. If you feel submitting numbers as strings is the way to go here, mark this comment resolved.

File: apps/sertra/pages/create.tsx [hunk 10]
```diff
@@ -17,13 +17,45 @@ export const getStaticProps: GetStaticProps = async () => {
...
+  const onSubmit: SubmitHandler<Inputs> = (data) => {
+    const prepped_data = {
+      name: data.serverName,
+      description: data.description,
+      community: "v-rising",
+      connection_data: data.connectionInfo,
+      // TODO: Populate this with mod ids etc
+      mods: ["1", "2", "3"],
+      is_pvp: data.mode === "pvp" ? true : false,
+      requires_password: data.isPasswordProtected ? true : false,
```
Comment (anttimaki) [inline apps/sertra/pages/create.tsx:48]: If `data` now implements `Input` interface, then `data.isPasswordProtected` is already a boolean and this ternary operator is redundant.

**Review by anttimaki — COMMENTED** [review]
Few more comments

File: apps/sertra/pages/create.tsx (hunk 10 above)
Comment (Oksamies) [reply apps/sertra/pages/create.tsx:48]: Comment about mods list of string was answer for both 👍 Also good that you pointed out that password field thing, didn't realize that some how 👌

**Review by anttimaki — APPROVED** [review]
Well the RHF import still isn't in alphabetical order within the import block but I guess that's something that should be handled by linter.

**Pull request merged**